pkg/rb2wf/converter.go

```diff
@@ -0,0 +1,586 @@
+	tsConfig string
+
+	//go:embed templates/blocks/unimplemented.ts.tmpl
+	unimplementedTemplate string
+)
+
+type RunbookConverter struct {
+	client     api.APIClient
+	entrypoint string
+	outputDir  string
+}
+
+func NewRunbookConverter(
+	client api.APIClient,
+	outputDir string,
+	entrypoint string,
+) *RunbookConverter {
+	return &RunbookConverter{
+		client:     client,
+		entrypoint: entrypoint,
+		outputDir:  outputDir,
+	}
+}
+
+func (r *RunbookConverter) Convert(ctx context.Context, runbookSlug string) error {
+	logger.Step("Fetching details of runbook %s", runbookSlug)
+	runbookInfo, err := r.getRunbookInfo(ctx, runbookSlug)
+	if err != nil {
+		return err
+	}
+
+	workflowParams := map[string]map[string]interface{}{}
+	for _, param := range runbookInfo.runbook.Runbook.Parameters {
+		workflowParams[param.Slug] = map[string]interface{}{
+			"name": param.Name,
+			"slug": param.Slug,
+			"type": typeToWorkflowType(param.Type),
+		}
+
+		if param.Default != nil {
+			workflowParams[param.Slug]["default"] = param.Default
+		}
+	}
+
+	envVars := map[string]map[string]interface{}{}
+	configs := map[string]string{}
+
+	for _, configVar := range runbookInfo.runbook.Runbook.TemplateSession.Configs {
+		envVars[configVar.NameTag] = map[string]interface{}{
+			"config": configVar.NameTag,
+		}
+		configs[configVar.NameTag] = fmt.Sprintf(
+			"%sprocess.env['%s']%s",
+			noQuoteStart,
+			configVar.NameTag,
+			noQuoteEnd,
+		)
+	}
+
+	logger.Step("Getting resources associated with the runbook")
+	resources, err := r.getResources(ctx, runbookInfo.blocks.Blocks)
+	if err != nil {
+		return err
+	}
+	resourceSlugs := []string{}
+	for _, resource := range resources {
+		resourceSlugs = append(resourceSlugs, resource.Slug)
+	}
+
+	sort.Slice(resourceSlugs, func(a, b int) bool {
+		return resourceSlugs[a] < resourceSlugs[b]
+	})
+
+	bodyChunks := []string{}
+
+	for _, block := range runbookInfo.blocks.Blocks {
+		logger.Step("Converting block %s to code", block.Slug)
+		blockStr, err := r.blockToString(ctx, block, resources)
+		if err != nil {
+			return err
+		}
+		bodyChunks = append(bodyChunks, blockStr)
+	}
+
+	logger.Step("Creating %s", r.entrypoint)
+	mainTsStr, err := applyTemplate(mainTemplate, struct {
+		Body        string
+		Configs     interface{}
+		Constraints interface{}
+		EnvVars     interface{}
+		Name        string
+		Parameters  interface{}
+		Resources   []string
+		RunbookID   string
+		RunbookSlug string
+		Slug        string
+	}{
+		Body:        strings.Join(bodyChunks, "\n"),
+		Configs:     configs,
+		Constraints: constraintsToMap(runbookInfo.runbook.Runbook.TemplateSession.Constraints),
+		EnvVars:     envVars,
+		Name:        fmt.Sprintf("%s (workflow)", runbookInfo.runbook.Runbook.Name),
+		Parameters:  workflowParams,
+		Resources:   resourceSlugs,
+		RunbookID:   runbookInfo.runbook.Runbook.ID,
+		RunbookSlug: runbookInfo.runbook.Runbook.Slug,
+		Slug:        fmt.Sprintf("%s_workflow", runbookInfo.runbook.Runbook.Slug),
```

Comment:
Runbook and task slugs are not exclusive, so we can reuse it here.

```suggestion
		Slug:        runbookInfo.runbook.Runbook.Slug,
```

Reply:
Fixed in https://github.com/airplanedev/cli/pull/823.